### PR TITLE
`<@argument>` Parsing error: Cannot read properties of null (reading 'index')

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

File: src/parser/transforms.js [hunk 1]
```diff
@@ -268,7 +268,7 @@ module.exports.preprocessGlimmerTemplates = function preprocessGlimmerTemplates(
         let codeSlice = code.slice(...n.range);
         for (const part of n.tag.split('.')) {
           let pattern = `\\b${part}\\b`;
-          if (part.startsWith(':')) {
+          if (part.startsWith(':') || part.startsWith('@')) {
```
Comment (patricklx) [inline src/parser/transforms.js:271]: I should probably extend this to any special character?
A tag could also be `_Foo`

Comment (patricklx) [reply src/parser/transforms.js:271]: Or just use indexOf, that should work...

Comment (NullVoxPopuli) [reply src/parser/transforms.js]: Difference is _Foo could be defined locally

File: src/parser/transforms.js [hunk 2]
```diff
@@ -295,8 +292,11 @@ module.exports.preprocessGlimmerTemplates = function preprocessGlimmerTemplates(
         n.params = [];
       }
       if ('blockParams' in n && n.parent) {
-        let part = code.slice(...n.parent.range);
-        let start = n.parent.range[0];
+        // for blocks {{x as |b|}} the block range does not contain the block params...
+        // for element tag it does <x as b />
```
Comment (NullVoxPopuli) [inline src/parser/transforms.js:296]:
```suggestion
        // for element tag it does <x as |b|>
```

**Pull request merged**